src/libsync/propagateupload.cpp

```diff
@@ -21,6 +22,9 @@
 #include "utility.h"
 #include "filesystem.h"
 #include "propagatorjobs.h"
+#include "transmissionchecksumvalidator.h"
+#include "configfile.h"
```

Comment:
Oh no :-( a dependency in the sync engine from the config file.

Maybe there should be a syncengine->setChecsumAlgo instead.